components/pedigree/resources/src/main/resources/pedigree/model/import.js

```diff
@@ -15,137 +17,416 @@ define([
     PedigreeImport.prototype = {
     };
 
-    /*PedigreeImport.SUPORTED_FORMATS = {
-      PED:                    1,      // standard .PED format. Can only import family structure, gender and the affected status
-      PHENOTIPS_GRAPH:        2,      // Phenotips pedigree format, whithout positioning information (needs to be laid out automaticaly)
-      PHENOTIPS_INTERNAL_OLD: 3       // Phenotips internal format used during development and in test cases (to be replaced)
-    };
-
-    PedigreeImport.autodetectFormat = function(input) {
+    /**
+     * Returns the current version of the Full JSON represenation
+     */
+    PedigreeImport.getCurrentFullJSONVersion = function() {
+        return "1.0";
+    },
 
-    }*/
+    /**
+     * Returns the set of all supported Full JSON versions which can be imported
+     */
+    PedigreeImport.getSupportedPedigreeJSONVersions = function() {
+        var versions = {};
+        versions[PedigreeImport.getCurrentFullJSONVersion()] = true;
+        return versions;
+    },
 
-    PedigreeImport.initFromPhenotipsInternal = function(inputG)
+    PedigreeImport.initFromFullJSON = function(inputText)
     {
-        // note: serialize() produces the correct input for this function
+        var KEY_PROBAND = "proband";
+        var KEY_MEMBERS = "members";
+        var KEY_RELATIONSHIPS = "relationships";
+        var KEY_RELATIONSHIPS_MEMBERS = "members";
+        var KEY_RELATIONSHIPS_CHILDREN = "children";
+        var KEY_LAYOUT = "layout";
+        var KEY_PROPERTIES_PHENOTIPSID = "id";
+        var KEY_PROPERTIES_GENDER = "sex";
 
-        var newG = new BaseGraph();
+        try {
+            var input = JSON.parse(inputText);
+        } catch(err) {
+            throw "Unable to import pedigree: input is not a valid pedigree JSON: " + err.message;
+        }
 
-        var nameToId = {};
+        if (!input.hasOwnProperty("JSON_version")
```

Comment:
why not create `KEY_PROPERTIES_...` for other key strings, like `JSON_version`, `pedigreeProperties`, `phenotipsId`, `gender`, etc?